### PR TITLE
Code with Suggested Changes

### PR DESCRIPTION
**Pull request by skthinks**

### CODE REVIEW

File: movie.sh [hunk 1]
```diff
@@ -1,55 +1,58 @@
...
-  elif [ $ctr -eq 1 ]
-  then
-        rating="$rating$var"
-  else
-        continue
-  fi
-done < "$input"
-rm mov.txt
-rm submov.txt
+        tput bold
```
Comment (vishesh92) [inline movie.sh]: https://google.github.io/styleguide/shell.xml#Indentation

The indentation is different in every block.

File: movie.sh [hunk 2]
```diff
@@ -1,55 +1,58 @@
...
+        tput setaf $2
+        echo $1
+        tput sgr 0
+        return 0
+}
+
+mkdir tmp
+function get_rating()
+{
+response=$( curl -s  "http://www.omdbapi.com/?t=$1&y=&plot=short&r=xml" | grep -o imdbRating=".*" | cut -c 13,14,15)
```
Comment (vishesh92) [inline movie.sh]: https://google.github.io/styleguide/shell.xml#Line_Length_and_Long_Strings
Line length

File: movie.sh [hunk 3]
```diff
@@ -1,55 +1,58 @@
...
-rm mov.txt
-rm submov.txt
+        tput bold
+        tput setaf $2
+        echo $1
+        tput sgr 0
+        return 0
+}
+
+mkdir tmp
```
Comment (vishesh92) [inline movie.sh]: http://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/tmp.html

File: README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+The application takes in a file path as input
+
+It is assumed that the user specifies a directory filled with movies
```
Comment (vaidik) [inline README.md:3]: Nice README file. To format your READMEs better, there is a syntax called Markdown syntax. Check it out - you can make your README look prettier.

File: movie.sh (hunk 3 above)
Comment (vaidik) [inline movie.sh]: We didn't mean creating a tmp directory. What we meant is that there is already a directory at `/tmp` for this purpose. You can use that.

File: movie.sh [hunk 5]
```diff
@@ -1,55 +1,58 @@
...
-ls movies > movlist.txt
-sed 's/_/./g' movlist.txt > mov2.txt
-input2="mov2.txt"
-echo Rating '     '  Novie >> out.txt
-while IFS= read -r var2
+#There should be exactly one argument i.e the address path
+if [ $# != 1 ]
+then
+	add_style "This application takes one argument : Movie Dorectory" 1
+	exit 1
```
Comment (vaidik) [inline movie.sh]: Nice!

File: movie.sh [hunk 6]
```diff
@@ -1,55 +1,58 @@
...
+fi
+#Fetching Movies and removing 
+ls $1 > tmp/movlist.txt
```
Comment (vaidik) [inline movie.sh]: Still using files? How can we get rid of them?

File: movie.sh (hunk 2 above)
Comment (vaidik) [inline movie.sh]: No indendation inside a function?

File: movie.sh [hunk 7]
```diff
@@ -1,55 +1,58 @@
...
+        echo $1
+        tput sgr 0
+        return 0
+}
+
+mkdir tmp
+function get_rating()
+{
+response=$( curl -s  "http://www.omdbapi.com/?t=$1&y=&plot=short&r=xml" | grep -o imdbRating=".*" | cut -c 13,14,15)
 #Convert String to Float
```
Comment (vishesh92) [inline movie.sh]: All comments should be on the same level of indentation as the code it refers to. All comments must start with `#` (i.e. a hash symbol and a space). e.g.
`# Convert String to Float`

File: movie.sh (hunk 2 above)
Comment (vaidik) [inline movie.sh]: Space not required after the `$(`.

Also you should break really long lines into multiple lines. This is how you can do:

```
response=$(curl -s "http://www.omdbapi.com/?t=$1&y=&plot=short&r=xml" \
    | grep -o imdbRating=".*" | cut -c 13,14,15)
```

Comment (vaidik) [inline movie.sh]: Notice how readable this has become now.

File: movie.sh [hunk 8]
```diff
@@ -1,55 +1,58 @@
...
-exit 1
+clear
+sort -k 1 -r tmp/out.txt
+tput setaf 1
+tput bold
+printf  "\n\nN/A: The application was unable to fetch your movie. We regret this deeply\n"
+tput sgr 0
+rm -rf tmp
+exit 0
+
```
Comment (vishesh92) [inline movie.sh]: Blank lines

**Pull request merged**